src/logcollector/read_snortfull.c

```diff
@@ -51,21 +51,21 @@ void *read_snortfull(logreader *lf, int *rc, int drop_it) {
         /* First part of the message */
         if (p == NULL) {
             if (strncmp(str, "[**] [", 6) == 0) {
-                strncpy(f_msg, str, OS_MAXSTR);
-                f_msg_size -= strlen(str) + 1;
+                snprintf(f_msg, sizeof(f_msg), "%s", str);
+                f_msg_size -= strlen(str);
                 p = one;
             }
         } else {
             if (p == one) {
                 /* Second line has the [Classification: */
                 if (strncmp(str, "[Classification: ", 16) == 0) {
                     strncat(f_msg, str, f_msg_size);
-                    f_msg_size -= strlen(str) + 1;
+                    f_msg_size -= strlen(str);
                     p = two;
                 } else if (strncmp(str, "[Priority: ", 10) == 0) {
                     strncat(f_msg, "[Classification: Preprocessor] "
                             "[Priority: 3] ", f_msg_size);
-                    f_msg_size -= strlen(str) + 1;
+                    f_msg_size -= strlen("[Classification: Preprocessor] [Priority: 3] ");
```

Comment:
Use compile-time expressions for constants.

```suggestion
                    f_msg_size -= sizeof("[Classification: Preprocessor] [Priority: 3] ") - 1;
```